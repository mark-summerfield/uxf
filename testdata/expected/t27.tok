TClassBegin ttype=<Files>
  Field Str("Kind")
  Field Str("Filename")
TClassEnd
MapBegin ktype=str
  Str("General")
  MapBegin ktype=str # Map comment
    Str("saved")
    Date(2022-03-21)
    Str("autosave")
    Bool(true)
    Str("historysize")
    Int(-35)
    Str("volume")
    Real(0.7)
    Str("files")
    TableBegin ttype=<Files> # Table comment
      Str("current")
      Str("/home/mark/app/rs/tlm/PlaylistsTest.tlm")
      Str("recent1")
      Str("/home/mark/app/rs/tlm/PlaylistsTest.tlm")
      Str("recent2")
      Str("/home/mark/data/playlists-all.tlm")
    TableEnd
  MapEnd
  Str("Window")
  MapBegin ktype=str
    Str("x")
    Int(383)
    Str("y")
    Int(124)
    Str("width")
    Int(590)
    Str("height")
    Int(536)
    Str("scale")
    Real(1.1)
  MapEnd
  Str("Magic")
  Bytes( (:1F8B11FF0C:) )
  Str("Nested Map")
  MapBegin ktype=str vtype=list
    Str("Classical")
    ListBegin
      Int(5)
      Bool(true)
    ListEnd
    Str("Modern Instrumental")
    ListBegin
      Int(4)
      Bool(false)
    ListEnd
    Str("New Acquistions")
    ListBegin
      Int(1)
      Bool(false)
    ListEnd
    Str("Nested List of Lists")
    ListBegin vtype=<list> # List comment
      ListBegin
        Int(5)
        Str("Classical & <Special>")
        Bool(true)
      ListEnd
      ListBegin
        Int(4)
        Str("Modern Instrumental")
        ?
      ListEnd
      ListBegin
        Int(1)
        Str("New Acquistions")
        Bool(false)
      ListEnd
    ListEnd
  MapEnd
MapEnd
Eof
----------------------------------------
