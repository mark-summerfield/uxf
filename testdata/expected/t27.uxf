uxf 1.0
= Files Kind Filename
{str
  <General> { #<Map comment> str
    <saved> 2022-03-21
    <autosave> yes
    <historysize> -35
    <volume> 0.7
    <files> ( #<Table comment> Files
      <current> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
      <recent1> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
      <recent2> </home/mark/data/playlists-all.tlm>
    )
  }
  <Window> { str
    <x> 383
    <y> 124
    <width> 590
    <height> 536
    <scale> 1.1
  }
  <Magic> (:1F8B11FF0C:)
  <Nested Map> { str list
    <Classical> [5 yes]
    <Modern Instrumental> [4 no]
    <New Acquistions> [1 no]
    <Nested List of Lists> [ #<List comment> list
      [5 <Classical &amp; &lt;Special&gt;> yes]
      [4 <Modern Instrumental> ?]
      [1 <New Acquistions> no]
    ]
  }
}
