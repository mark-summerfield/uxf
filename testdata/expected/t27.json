{
  "UXF^ttypes": [
    {
      "UXF^ttype": {
        "name": "Files",
        "fields": {
          "Kind": null,
          "Filename": null
        }
      }
    }
  ],
  "UXF^data": {
    "UXF^map": {
      "comment": null,
      "ktype": "str",
      "vtype": null,
      "map": {
        "General": {
          "UXF^map": {
            "comment": "Map comment",
            "ktype": "str",
            "vtype": null,
            "map": {
              "autosave": true,
              "files": {
                "UXF^table": {
                  "name": "Files",
                  "comment": "Table comment",
                  "fields": {
                    "Kind": null,
                    "Filename": null
                  },
                  "records": [
                    [
                      "current",
                      "/home/mark/app/rs/tlm/PlaylistsTest.tlm"
                    ],
                    [
                      "recent1",
                      "/home/mark/app/rs/tlm/PlaylistsTest.tlm"
                    ],
                    [
                      "recent2",
                      "/home/mark/data/playlists-all.tlm"
                    ]
                  ]
                }
              },
              "historysize": -35,
              "saved": {
                "UXF^date": "2022-03-21"
              },
              "volume": 0.7
            }
          }
        },
        "Magic": {
          "UXF^bytes": "1F8B11FF0C"
        },
        "Nested Map": {
          "UXF^map": {
            "comment": null,
            "ktype": "str",
            "vtype": "list",
            "map": {
              "Classical": [
                5,
                true
              ],
              "Modern Instrumental": [
                4,
                false
              ],
              "Nested List of Lists": {
                "UXF^list": {
                  "comment": "List comment",
                  "vtype": "list",
                  "list": [
                    [
                      5,
                      "Classical & <Special>",
                      true
                    ],
                    [
                      4,
                      "Modern Instrumental",
                      null
                    ],
                    [
                      1,
                      "New Acquistions",
                      false
                    ]
                  ]
                }
              },
              "New Acquistions": [
                1,
                false
              ]
            }
          }
        },
        "Window": {
          "UXF^map": {
            "comment": null,
            "ktype": "str",
            "vtype": null,
            "map": {
              "height": 536,
              "scale": 1.1,
              "width": 590,
              "x": 383,
              "y": 124
            }
          }
        }
      }
    }
  }
}